packages/shared_preferences/shared_preferences_android/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -133,12 +135,18 @@ public void onDetachedFromEngine(@NonNull FlutterPlugin.FlutterPluginBinding bin
   }
 
   // Gets all shared preferences, filtered to only those set with the given prefix.
+  // Optionally filtered also to only those items in the optional [allowList].
   @SuppressWarnings("unchecked")
-  private @NonNull Map<String, Object> getAllPrefs(@NonNull String prefix) throws RuntimeException {
+  private @NonNull Map<String, Object> getAllPrefs(
+      @NonNull String prefix, @Nullable List<String> allowList) throws RuntimeException {
     Map<String, ?> allPrefs = preferences.getAll();
     Map<String, Object> filteredPrefs = new HashMap<>();
+    Set<String> allowSet = null;
+    if (allowList != null) {
+      allowSet = new HashSet<>(allowList);
+    }
     for (String key : allPrefs.keySet()) {
-      if (key.startsWith(prefix)) {
+      if (key.startsWith(prefix) && (allowSet == null || allowSet.contains(key))) {
```

Comment:
This aligns with the documentation but I would expect this function to actually or these values. 
As a developer I want getAllPrefs to Give me everything with my custom prefix and a list of legacy values I used before.

Reply:
If you go with the OR approach then I think you should also make the new param an optional named parameter.